packages/field-plugin/src/createFieldPlugin/createFieldPlugin.ts

```diff
@@ -6,14 +6,13 @@ import { FieldPluginResponse } from './FieldPluginResponse'
 import { createPluginMessageListener } from './createPluginActions/createPluginMessageListener'
 import { sandboxUrl } from './sandboxUrl'
 
-export type CreateFieldPlugin = (
-  onUpdate: (state: FieldPluginResponse) => void,
-) => () => void
-
 /**
  * @returns cleanup function for side effects
  */
-export const createFieldPlugin: CreateFieldPlugin = (onUpdateState) => {
+export const createFieldPlugin = <Content>(
+  onUpdateState: (state: FieldPluginResponse) => void,
+  parseContent: (content: unknown) => Content,
```

Comment:
Should this be mandatory?

If we allow it to be optional, there is a greater risk that the user encounter errors due to unexpected formatting of the content. If we make `parseContent` mandatory, they can still escape the validation by passing the identity function.